src/x509.lisp

```diff
@@ -297,3 +297,16 @@ designator for the digest algorithm to use (it defaults to SHA-1)."
                  :message "failed to compute fingerprint of certificate"
                  :queue (read-ssl-error-queue))))
       fingerprint)))
+
+(defun x509-cert-from-pem (pem)
+  (with-bio-input-from-string (bio pem) (pem-read-x509 bio 0 0 0))
+  pem)
```

Comment:
Returning the original input string? Most likely last line is not needed.

Reply:
Right. I have it fixed locally, but I keep additional changes in x509 until core is fixed. I consider I would remove this from this patch/final pull request and left only the core bio fix - that is itself  larger than I originally thought. But your call.